### PR TITLE
Deprecate Identity's Formstack features

### PR DESCRIPTION
**Pull request by sunnyclimber456**
## What is the value of this and can you measure success?

This PR deprecates rendering Formstack forms for an authenticated user in the Identity application. This functionality was originally introduced in e329b290bfe2ac03cfd6e43710f041659221d4c9, but we have not used Formstack in this way for some time. Deprecating these features will simplify the migration to Okta.

## What does this change?

This PR deprecates the following routes, along with any modules associated with them that are no longer used.

- **Deprecate:** `/form/:formReference`  - e.g https://profile.theguardian.com/form/1645439-tOfqZ3Kxj2
  - this route will now return a 404
- **Deprecate**: `/form/embed/:formReference` - e.g https://profile.theguardian.com/form/embed/1645439-tOfqZ3Kxj2
  - this route will now return a 404

This PR keeps the following route, because logs show it is still being used. We should replicate it in another application and add a redirect, as well as inform people to use the new one.

- **Keep**: [`/form/complete`](https://profile.theguardian.com/form/complete)

## What does this _not_ change?

This doesn't change any Formstack embeds _outside_ of the Identity application. For [example](https://www.theguardian.com/lifeandstyle/2023/jun/09/my-boyfriends-aggressive-dog-has-stalled-our-plans-to-move-in-together-how-can-we-move-forwards?dcr=false), Formstack embeds in regular articles are unchanged.

### CODE REVIEW

File: common/app/common/configuration.scala [hunk 1]
```diff
@@ -707,11 +707,6 @@ class GuardianConfiguration extends GuLogging {
     lazy val apiKey = configuration.getMandatoryStringProperty("riffraff.apikey")
   }
 
-  object formstack {
-    val url = configuration.getMandatoryStringProperty("formstack.url")
-    val identityOauthToken = configuration.getMandatoryStringProperty("formstack.oauthToken")
-  }
```
Comment (jamesgorrie) [inline common/app/common/configuration.scala:713]: Note on cleanup - we should remove these from SSM

File: identity/app/controllers/FormstackController.scala [hunk 2]
```diff
@@ -1,71 +1,31 @@
...
-            }
-          }
-          .getOrElse {
-            Future.successful(NotFound(views.html.formstack.formstackFormNotFound(page)))
-          }
-      } else {
-        logger.info(s"formstack switched off, attempt to access $formReference failed")
-        Future.successful(NotFound(views.html.errors._404()))
-      }
+  def formstackForm(formReference: String): Action[AnyContent] =
```
Comment (jamesgorrie) [inline identity/app/controllers/FormstackController.scala:22]: I wonder if we could deprecate this more, and copy the HTML & CSS into an S3 bucket and just serve from there? 

Thought for later about how we deprecate routes completely.

Comment (jamesgorrie) [inline identity/app/controllers/FormstackController.scala:22]: Could we remove `formReference`

**Review by jamesgorrie — APPROVED** [review]
Screenshot 2023-07-18 at 14 27 38

Yes please

File: identity/app/controllers/FormstackController.scala (hunk 2 above)
Comment (georgeblahblah) [reply identity/app/controllers/FormstackController.scala:22]: interesting idea, this is now essentially a static page. I wonder if we can use frontend's existing static page handling in the meantime -- I'll investigate!

Comment (georgeblahblah) [reply identity/app/controllers/FormstackController.scala:22]: annoyingly I don't think we can -- Play complains if we specify a route parameter which we don't consume, and I think we need to keep the route parameter for now.

We could rename the parameter to `unused` or something similar - what do you think?

**Pull request merged**